pgsqltoolsservice/connection/connection_service.py

```diff
@@ -300,16 +294,25 @@ def _close_connections(connection_info: ConnectionInfo, connection_type=None):
                 # Ignore errors when disconnecting
                 pass
         return True
+    
+    def _get_provider_name(self, owner_uri):
+        # Get info about this connection's provider
+        if not self._service_provider.provider_name:
+            provider_name = get_attribute_value(owner_uri, "providerName")
+            self._service_provider.provider_name = provider_name
+            return provider_name
+        else:
+            return self._service_provider.provider_name
 
 
 def _build_connection_response(connection_info: ConnectionInfo, connection_type: ConnectionType) -> ConnectionCompleteParams:
     """Build a connection complete response object"""
     connection = connection_info.get_connection(connection_type)
-    dsn_parameters = connection.get_dsn_parameters()
+    dsn_parameters = connection.dsn_parameters
```

Comment:
Do we need this as dsn_parameters ? Or just flatten out all the properties in the connection class ?

Reply:
I ended up flattening out the relevant properties in the connection class.